### ISSUE TITLE
Pagemap and mm.img file

### ISSUE THREAD

**Issue by shubham95** [checkpoint-restore/criu#1101]
mm-(pid).img       file contains info of VMAs 
pages.img            file contains raw data of pages
pagemap.img       file contains starting virtual address and nr of pages g 

What is use of pagemap file because during comaprison between mm and pagemap file it is not exatly same(means in pagemap file we can find end address by (start address + nr_pages *PAGESIZE) )

how pagemap.img file help us to restore pages or it is just indexs to page.img file?

Can you give overview of pages restore in pre-dump case
How pages refered to parent directory?

**Comment by rppt** [comment]
Please look at https://criu.org/Memory_dumping_and_restoring and related pages.

**Comment by shubham95** [comment]
Ok, i went through these articles and I got clarification :)
I am trying to understand the codebase of restorer mainly 
in which I am not able find how pages are restored.


What I figured out till now.
after fork your process starts with function restore_task_with_children():
my interest is in function    prepare_mappings()
In prepare_mappings() 
we call open_page_read() it fills the pr(page_read) struct
premap_priv_vmas
restore_priv_vma_content
In restore_priv_vma_content we put requests in pagemap_enqueue_iovec

I am not getting where actual restoring of pages happens
is it done in restorer blob?

which function do i have look more?
